### DISCUSSION TITLE
Silent Bacth Job - Bulk Export

### DISCUSSION THREAD

**Discussion by wmti-admin** [Micke-K/IntuneManagement#268] [Q&A] [answered]
I was hoping someone could expand a little upon the Silent Batch Job portion of the documentation, specifically what's required to do a silent bulk export. I'd like to setup a scheduled task to accomplish this on a regular basis. After doing some experimenting, it's not quite clear to me as to what is required for the -SilentBatchFile argument of the command. Thanks

**Comment by wmti-admin** [comment]
After doing some more digging I see that the BulkExport.json settings file is what is required. The documentation mentions that the tool will generate this file by default, however I cannot find it after performing a manual Bulk Export. How is this file generated?

**Reply by wmti-admin** [reply, marked as answer]
Problem solved. In the Bulk Import/Export settings menu of the GUI there is a 'Save' button in the bottom left corner that will export the selected settings into a .json file. This is the file that needs to be referenced in the silent execution command.

**Reply by Micke-K** [reply]
Hello!

Sorry for not answering. It's been crazy busy lately. 

Let me know if you have any other questions. 

Cheers!

**Reply by wmti-admin** [reply]
Hi Micke-K,

Thanks for your reply.

I do have another question related to this topic. I've been trying to get the authentication to work silently but am having no luck. 

When I execute the app silently, I receive the following error on each API call: 

**_Failed to invoke MS Graph with URL. The remote server returned an error: (403) Forbidden._** 

I have an app registration created on my tenant that has delegated permissions assigned to the required MS Graph permissions. I have created a secret and am referencing the value in the -Secret switch of the command. I have also tried executing the command with only the -Silent -SilentBatchFile and -TenantId switches to authenticate through the settings stored in the app but am receiving the same error. I can export through the app GUI just fine so I know my settings are correct.

I've tried to use the _Login with App in UI (Preview)_ setting but after restarting the app, I receive the following error when clicking the login button, even though all of these fields have been set in both Settings and Tenant Settings menu:

**_Azure AppId, Tenant ID and Secret/Cert must be specified for App logins_**

Any guidance is greatly appreciated.

Thanks

**Reply by Micke-K** [reply]
Hello,

"I have an app registration created on my tenant that has **delegated permissions** assigned to the required MS Graph permissions"

You need to assign Application Permissions for the silent batch job to work with an App Registration.

Cheers
